<launch>
<param name ="/use_sim_time" value="true"/>
<arg name="gcs_url" default=""/>
<arg name="pluginlists_yaml" default="$(find mavros)/launch/px4_pluginlists.yaml" />
<arg name="config_yaml" default="$(find mavros)/launch/px4_config.yaml" />

<group ns="camila0">
  <arg name="ID" value="0"/>
	<node name="mavros" pkg="mavros" type="mavros_node" output="screen">
		
		<param name="fcu_url" value="udp://:14540@localhost:14557"/>
		<param name="gcs_url" value="$(arg gcs_url)" />
		<param name="target_system_id" value="1" />
		<param name="target_component_id" value="1" />
                <rosparam command="load" file="$(arg pluginlists_yaml)" />
                <rosparam command="load" file="$(find swarm_ctrl_pkg)/px4-configs/fcu_config/px4_config_$(arg ID).yaml" />

                <param name="local_position/tf/send" value="true" />
                <param name="local_position/tf/child_frame_id" value="camila$(arg ID)" />
                <param name="global_position/tf/send" value="false" />
                <param name="global_position/tf/child_frame_id" value="camila$(arg ID)" />
	</node>

</group>
<group ns="camila1">
  <arg name="ID" value="1"/>
	<node name="mavros" pkg="mavros" type="mavros_node" output="screen">
		
		<param name="fcu_url" value="udp://:14541@127.0.0.1:14558"/>
		<param name="gcs_url" value="$(arg gcs_url)" />
		<param name="target_system_id" value="1" />
		<param name="target_component_id" value="1" />
                <rosparam command="load" file="$(arg pluginlists_yaml)" />
                <rosparam command="load" file="$(find swarm_ctrl_pkg)/px4-configs/fcu_config/px4_config_$(arg ID).yaml" />
                <param name="local_position/tf/send" value="true" />
                <param name="local_position/tf/child_frame_id" value="camila$(arg ID)" />
                <param name="global_position/tf/send" value="false" />
                <param name="global_position/tf/child_frame_id" value="camila$(arg ID)" />
	</node>
</group>

<group ns="camila2">
  <arg name="ID" value="2"/>
	<node name="mavros" pkg="mavros" type="mavros_node" output="screen">
		
		<param name="fcu_url" value="udp://:14542@127.0.0.1:14559"/>
		<param name="gcs_url" value="$(arg gcs_url)" />
		<param name="target_system_id" value="1" />
		<param name="target_component_id" value="1" />
                <rosparam command="load" file="$(arg pluginlists_yaml)" />
                <rosparam command="load" file="$(find swarm_ctrl_pkg)/px4-configs/fcu_config/px4_config_$(arg ID).yaml" />
                <param name="local_position/tf/send" value="true" />
                <param name="local_position/tf/child_frame_id" value="camila$(arg ID)" />
                <param name="global_position/tf/send" value="false" />
                <param name="global_position/tf/child_frame_id" value="camila$(arg ID)" />
	</node>
</group>

<group ns="camila3">
  <arg name="ID" value="3"/>
	<node name="mavros" pkg="mavros" type="mavros_node" output="screen">
		
                <param name="fcu_url" value="udp://:14543@127.0.0.1:14539"/>
		<param name="gcs_url" value="$(arg gcs_url)" />
		<param name="target_system_id" value="1" />
		<param name="target_component_id" value="1" />
                <rosparam command="load" file="$(arg pluginlists_yaml)" />
                <rosparam command="load" file="$(find swarm_ctrl_pkg)/px4-configs/fcu_config/px4_config_$(arg ID).yaml" />
                <param name="local_position/tf/send" value="true" />
                <param name="local_position/tf/child_frame_id" value="camila$(arg ID)" />
                <param name="global_position/tf/send" value="false" />
                <param name="global_position/tf/child_frame_id" value="camila$(arg ID)" />
	</node>
</group>

<group ns="camila4">
  <arg name="ID" value="4"/>
        <node name="mavros" pkg="mavros" type="mavros_node" output="screen">

                <param name="fcu_url" value="udp://:14544@127.0.0.1:14538"/>
                <param name="gcs_url" value="$(arg gcs_url)" />
                <param name="target_system_id" value="1" />
                <param name="target_component_id" value="1" />
                <rosparam command="load" file="$(arg pluginlists_yaml)" />
                <rosparam command="load" file="$(find swarm_ctrl_pkg)/px4-configs/fcu_config/px4_config_$(arg ID).yaml" />
                <param name="local_position/tf/send" value="true" />
                <param name="local_position/tf/child_frame_id" value="camila$(arg ID)" />
                <param name="global_position/tf/send" value="false" />
                <param name="global_position/tf/child_frame_id" value="camila$(arg ID)" />
        </node>
</group>

<node name="cmd_node" pkg="swarm_ctrl_pkg" type="cmd_node" output="screen">
  <param name="takeoff_alt" value="4"/>
</node>

<node name="state_node" pkg="swarm_ctrl_pkg" type="state_node" output="screen">
</node>

<node name="set_point_node" pkg="swarm_ctrl_pkg" type="set_point_node" output="screen">
   <param name="formation" value="diamond"/>
   <param name="offset" value="0.0"/>
</node>

<node name="control_node" pkg="swarm_ctrl_pkg" type="control_node" output="screen">
</node>

</launch>
